cloudflare/resource_cloudflare_worker_kv.go

```diff
@@ -0,0 +1,127 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareWorkerKV() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareWorkersKVCreate,
+		Read:   resourceCloudflareWorkersKVRead,
+		Update: resourceCloudflareWorkersKVUpdate,
+		Delete: resourceCloudflareWorkersKVDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareWorkersKVImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"namespace_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareWorkersKVRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	namespaceID, key := parseId(d)
+
+	value, err := client.ReadWorkersKV(context.Background(), namespaceID, key)
+	if err != nil {
+		return errors.Wrap(err, "error reading workers kv")
+	}
+
+	if value == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("value", string(value))
+	return nil
+}
+
+func resourceCloudflareWorkersKVCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	namespaceID := d.Get("namespace_id").(string)
+	key := d.Get("key").(string)
+	value := d.Get("value").(string)
+
+	_, err := client.WriteWorkersKV(context.Background(), namespaceID, key, []byte(value))
+	if err != nil {
+		if err != nil {
+			return errors.Wrap(err, "error creating workers kv")
+		}
+	}
+
+	d.SetId(fmt.Sprintf("%s_%s", namespaceID, key))
+
+	log.Printf("[INFO] Cloudflare Workers KV Namespace ID: %s", d.Id())
+
+	return nil
```

Comment:
All methods that modify a remote resource should also perform a `Read` to ensure the remote state is maintained with the Terraform state. Check out the [filters](https://github.com/terraform-providers/terraform-provider-cloudflare/blob/master/cloudflare/resource_cloudflare_filter.go) resource for prior art. The [Terraform documentation](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#update-state-after-modification) also has a small write up on this.